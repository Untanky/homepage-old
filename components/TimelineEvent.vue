<template lang="pug">
  .timeline-event(:class="{ active: element.active }")
    .text-xl.font-medium {{element.date}}
    slot
</template>

<script lang="ts">
import Vue, { PropType } from 'vue';
import TimelineElement from '../src/TimelineElement';

export default Vue.extend({
  name: 'TimelineEvent',
  props: {
    element: {
      type: Object as PropType<TimelineElement>,
      required: true,
    },
  },
});
</script>

<style lang="postcss" scoped>
.timeline-event {
  @apply py-2;
  @apply pr-4;
  @apply pl-12;
  @apply inline-block;
  @apply relative;
}

.timeline-event::after {
  content: "";
  @apply w-3;
  @apply h-3;
  @apply block;
  @apply absolute;
  @apply bg-red-500;
  @apply rounded-full;
  @apply top-4;
  @apply left-6;
}

.active::before {
  content: "â€º";
  @apply w-3;
  @apply h-3;
  @apply block;
  @apply absolute;
  @apply text-red-500;
  @apply font-black;
  @apply text-2xl;
  @apply top-0.25;
  @apply left-2;
}
</style>
