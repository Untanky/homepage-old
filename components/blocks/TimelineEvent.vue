<i18n>
{
  "en": {
    "timeline": {
      "now": "now"
    }
  },
  "de": {
    "timeline": {
      "now": "zurzeit"
    }
  }
}
</i18n>

<template lang="pug">
  .timeline-event(:class="{ active: element.active }")
    .text-xl.font-medium {{ dateString }}
    slot
</template>

<script lang="ts">
import Vue, { PropType } from 'vue';
import { TimelineEntry } from '../src/components';

const DATE_FORMAT = { month: 'long', year: 'numeric' };

export default Vue.extend({
  name: 'TimelineEvent',
  props: {
    element: {
      type: Object as PropType<TimelineEntry>,
      required: true,
    },
  },
  computed: {
    dateString(): string {
      const fromString = this.element.from
        .setLocale(this.$i18n.locale)
        .toLocaleString(DATE_FORMAT);
      const toString = this.formatToString();
      const optionalHyphen = toString !== '' ? ' - ' : '';

      return `${fromString}${optionalHyphen}${toString}`;
    },
  },
  methods: {
    formatToString(): string {
      if (this.element.active) {
        return this.$i18n.t('timeline.now').toString();
      }

      if (this.element.to) {
        return this.element.to
          .setLocale(this.$i18n.locale)
          .toLocaleString(DATE_FORMAT);
      }

      return '';
    },
  },
});
</script>

<style lang="postcss" scoped>
.timeline-event {
  @apply py-2;
  @apply pr-4;
  @apply pl-12;
  @apply inline-block;
  @apply relative;
}

.timeline-event::after {
  content: "";
  @apply w-3;
  @apply h-3;
  @apply block;
  @apply absolute;
  @apply bg-gradient-to-r;
  @apply from-primary-500;
  @apply to-secondary-500;
  @apply rounded-full;
  @apply top-4;
  @apply left-6;
}

.active::before {
  content: "â€º";
  @apply w-3;
  @apply h-3;
  @apply block;
  @apply absolute;
  @apply text-primary-500;
  @apply font-black;
  @apply text-2xl;
  @apply top-0.25;
  @apply left-2;
}
</style>
