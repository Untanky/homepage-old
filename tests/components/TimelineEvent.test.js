/* eslint-disable no-undef */
// global describe, it, expect
import { mount } from '@vue/test-utils';
import TimelineEvent from '../../components/TimelineEvent.vue';

describe('TimelineEvent', () => {
  it('should render correctly when active', () => {
    const props = {
      element: {
        date: 'November 2020 - December 2020',
        active: true,
      },
    };

    const comp = mount(TimelineEvent, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should render correctly when inactive', () => {
    const props = {
      element: {
        date: 'November 2020 - December 2020',
        active: false,
      },
    };

    const comp = mount(TimelineEvent, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });
});
