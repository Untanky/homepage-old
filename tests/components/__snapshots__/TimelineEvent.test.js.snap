// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineEvent should render correctly when active 1`] = `
<div
  class="timeline-event active"
>
  <div
    class="text-xl font-medium"
  >
    November 2020 - December 2020
  </div>
</div>
`;

exports[`TimelineEvent should render correctly when inactive 1`] = `
<div
  class="timeline-event"
>
  <div
    class="text-xl font-medium"
  >
    November 2020 - December 2020
  </div>
</div>
`;
